File: java_junit5_idea/src/ClosedRange.java
public class ClosedRange {
    private final int higher;
    private final int lower;

    public ClosedRange(int lower, int higher) {
        this.lower = lower;
        this.higher = higher;
    }

    public String toString() {
        return "[" + lower + "," + higher + "]";
    }
}

File: csharp_mstest/csharp_mstest_test/ClosedRange.cs




namespace csharp_mstest_test
{
    internal class ClosedRange
    {
        private int lower;
        private int upper;

        public ClosedRange(int lower, int upper)
        {
            if(lower > upper)
            {
                throw new System.ArgumentException();
            }
            this.lower = lower;
            this.upper = upper;
        }

        internal int getLower()
        {
            return lower;
        }

        internal int getUpper()
        {
            return upper;
        }

        public override string ToString()
        {
            return "[" + lower + "," + upper + "]";
        }

        internal bool contains(int point)
        {
            return lower <= point && point <= upper;
        }

        internal bool contains(ClosedRange other)
        {
            return lower <= other.getLower() && other.getUpper() <= upper;
        }
    }
}